python-packages/fle_utils/videos.py

```diff
@@ -36,18 +38,18 @@ def download_video(youtube_id, download_path="../content/", download_url=OUTSIDE
     thumb_filepath = os.path.join(download_path, thumb_filename)
 
     try:
-        path, response = download_file(url, filepath, callback_percent_proxy(callback, end_percent=95))
+        response = download_file(url, filepath, callback_percent_proxy(callback, end_percent=95))
         if (
                 not os.path.isfile(filepath) or
-                "Content-Length" not in response or
-                not str(len(open(filepath, "rb").read())) == response["Content-Length"]):
+                "content-length" not in response.headers or
+                not len(open(filepath, "rb").read()) == int(response.headers['content-length'])):
             raise URLNotFound("Video was not found, tried: {}".format(url))
 
-        path, response = download_file(thumb_url, thumb_filepath, callback_percent_proxy(callback, start_percent=95, end_percent=100))
+        response = download_file(thumb_url, thumb_filepath, callback_percent_proxy(callback, start_percent=95, end_percent=100))
         if (
                 not os.path.isfile(thumb_filepath) or
-                "Content-Length" not in response or
-                not str(len(open(thumb_filepath, "rb").read())) == response["Content-Length"]):
+                "content-length" not in response.headers or
+                not len(open(thumb_filepath, "rb").read()) == int(response.headers['content-length'])):
```

Comment:
Expand and look at line 55-68, that's where all the error handling essentially takes place, even a catch-all-exceptions block :/